### PR TITLE
Update precedence of templates for 7.0

### PR DESCRIPTION
**Pull request by wtgodbe**
Part of https://github.com/dotnet/aspnetcore/issues/39731. Going to push an update to this PR that updates documentation as well (needed to create the PR first so I could link to it from the docs)

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json [hunk 1]
```diff
@@ -9,10 +9,10 @@
   "generatorVersions": "[1.0.0.0-*)",
   "description": "A project template for creating a Blazor server app that runs server-side inside an ASP.NET Core app and handles user interactions over a SignalR connection. This template can be used for web apps with rich dynamic user interfaces (UIs).",
   "groupIdentity": "Microsoft.Web.Blazor.Server",
-  "precedence": "8000",
-  "identity": "Microsoft.Web.Blazor.Server.CSharp.6.0",
+  "precedence": "9000",
+  "identity": "Microsoft.Web.Blazor.Server.CSharp.7.0",
   "shortName": "blazorserver",
-  "thirdPartyNotices": "https://aka.ms/aspnetcore/6.0-third-party-notices",
+  "thirdPartyNotices": "https://aka.ms/aspnetcore/7.0-third-party-notices",
```
Comment (wtgodbe) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json:15]: @mkArtakMSFT could you update/create this aka.ms link?

**Review by HaoK — APPROVED** [review]
Changes look good, sorta of related question, is there any kind of test we can add to guard against these precedence changes being incorrect?

**Comment by wtgodbe** [comment]
> Changes look good, sorta of related question, is there any kind of test we can add to guard against these precedence changes being incorrect?

I'm not sure, I wasn't aware of them until today. Short of waiting for an SDK that ingests them, maybe @MichalPavlik or @vlada-shubina would have an idea?

**Comment by HaoK** [comment]
Or maybe we just ask CTI to add a scenario where they have 6.0 and 7.0 installed, and run dotnet new, I believe that would catch it right since it would install the wrong version?  We could automate that but that's much harder for us to do with our tests and the versioning hell than just verifying this on real builds

**Comment by HaoK** [comment]
Yeah that's fine, basically we merge this as is, and then ask CTI to add a scenario for this to start testing once the changes are actually in

**Comment by vlada-shubina** [comment]
> > Changes look good, sorta of related question, is there any kind of test we can add to guard against these precedence changes being incorrect?
> 
> I'm not sure, I wasn't aware of them until today. Short of waiting for an SDK that ingests them, maybe @MichalPavlik  or @vlada-shubina  would have an idea?

We have an idea of enabling template validation via MSBuild tasks: https://github.com/dotnet/templating/issues/3828, however it might not meet the bar in near future. MSBuild tasks package is already in use in several repos for localization and we are onboarding others, so it's a matter of adding validation logic to them to report error/warnings in templates during the build. Detecting precedence / identity clashes is a bit tricky as in these scenarios error is over 2 or more packages, not individual templates inside same packages, but still detecting it is possible.

**Comment by dougbu** [comment]
@halter73 is there a general issue affecting acquisition of Windows build agents❔

**Comment by halter73** [comment]
I found another PR that was [similarly affected](https://dev.azure.com/dnceng/public/_build/results?buildId=1574065&view=results). I retriggered the canceled builds.

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json (hunk 1 above)
Comment (mkArtakMSFT) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json:15]: Just created. The link should be active soon.

**Pull request merged**